src/admin.rs

```diff
@@ -176,24 +177,26 @@ where
 
     let mut res = BytesMut::new();
     res.put(row_description(&columns));
-
-    for shard in 0..pool.shards() {
-        for server in 0..pool.servers(shard) {
-            let address = pool.address(shard, server);
-            let stats = match stats.get(&address.id) {
-                Some(stats) => stats.clone(),
-                None => HashMap::new(),
-            };
-
-            let mut row = vec![address.name(), config.user.name.clone()];
-
-            for column in &columns[2..columns.len() - 1] {
-                let value = stats.get(column.0).unwrap_or(&0).to_string();
-                row.push(value);
+    for (_, pool) in get_all_pools() {
+        let pool_config = pool.settings.clone();
```

Comment:
I think you can borrow the pool config here, e.g. `let pool_config = &pool.settings;`, but it doesn't really matter.